### PR TITLE
Update Newtonsoft.Json to 13.0.1

### LINKED ISSUE #11095: Update Newtonsoft.Json to 13.0.1
Update NuGet's dependency on Newtonsoft.Json to version 13.0.1

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11095

Regression? no

## Description

Update package version, and do some cleanup.

## PR Checklist

- [X] PR has a meaningful title
- [X] PR has a linked issue.
- [X] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [X] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [X] N/A

### CODE REVIEW

**Review by zkat — APPROVED** [review]

**Comment by zivkan** [comment]
blocked on https://github.com/dotnet/sdk/pull/19440

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj [hunk 1]
```diff
@@ -66,7 +66,7 @@
       <Link>Modules\NuGet\NuGet.PackageManagement.PowerShellCmdlets.dll-Help.xml</Link>
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
-    <Content Include="$(SolutionPackagesFolder)newtonsoft.json\9.0.1\lib\net45\Newtonsoft.Json.dll">
+    <Content Include="$(SolutionPackagesFolder)newtonsoft.json\13.0.1\lib\net45\Newtonsoft.Json.dll">
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: nit: You can use the newtonsoft version variable.

File: build/sign.proj [hunk 2]
```diff
@@ -23,7 +23,7 @@
       <FilesToSign Include="$(SolutionPackagesFolder)lucene.net\3.0.3\lib\NET40\Lucene.Net.dll">
         <Authenticode>3PartySHA2</Authenticode>
       </FilesToSign>
-      <FilesToSign Include="$(SolutionPackagesFolder)newtonsoft.json\9.0.1\lib\net45\Newtonsoft.Json.dll">
+      <FilesToSign Include="$(SolutionPackagesFolder)newtonsoft.json\13.0.1\lib\net45\Newtonsoft.Json.dll">
```
Comment (nkolev92) [inline build/sign.proj:26]: nit: You can use the newtonsoft version variable.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs [hunk 3]
```diff
@@ -492,6 +492,12 @@ private void CopyRestoreArtifacts(string artifactsDirectory, string pathToSdkInC
                         overwrite: true);
                 }
             }
+
+            // temp: delete once the .NET SDK ships Newtonsoft.Json 13.0.1 or higher
```
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs:496]: tracking issue?

**Review by nkolev92 — APPROVED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj (hunk 1 above)
Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: The MSBuild property with the version number is defined in `packages.targets` which is imported after this line. I don't feel like spending time moving imports around to get this working without causing problems.

File: build/sign.proj (hunk 2 above)
Comment (zivkan) [reply build/sign.proj:26]: This proj file doesn't currently import `package.targets`, and while it wouldn't be difficult to do so, I don't feel like spinning another build.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: Content is an item, but the version number is a property.
Property pass happens before the items pass, so the ordering wouldn't matter.

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs (hunk 3 above)
Comment (zivkan) [reply test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs:496]: https://github.com/NuGet/Home/issues/11135

I'm not going to spin another build just to change the comment.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj (hunk 1 above)
Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: Are you sure properties are evaluated before items?  I thought MSBuild was strictly "top down"

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: Yep, https://docs.microsoft.com/en-us/visualstudio/msbuild/build-process-overview?view=vs-2019

Comment (zivkan) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: I'll try to remember for the cleanup issue.

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj:69]: Only do that if you want to. 
I don't want these comments to add any additional work. 
It's nice to have, but really not that important in the grand scheme.

**Pull request merged**